src/main/java/com/sforce/async/BulkConnection.java

```diff
@@ -90,6 +90,9 @@
     public static final TypeMapper typeMapper = new TypeMapper();
     private static final JsonFactory factory = new JsonFactory(new ObjectMapper());
 
+    public BulkConnection() {
```

Comment:
The rest of your changes look ok. But I'm not ok merging this change in.

It breaks the contract that our connections require. All connections should have a connector config.